### ISSUE TITLE
Functional encoding is broken

### ISSUE THREAD

**Issue by nickgian** [NetworkVerification/nv#11]
The functional encoding seems to be broken.
When I try to run it on examples/fat8/fat8.nv NV crashes on renaming.

I made a branch with the last known commit where that did not happen:
https://github.com/princedpw/nv/tree/debugging-func

It's quite old, but maybe we can work backwards to see what went wrong. This seems like a bug in one of our transformations. Devon any insights?

**Comment by DKLoehr** [comment]
I did some digging and seem to have eliminated the crashes. However, the encoding seems to be giving incorrect results for batfish). It returns no counterexamples even though the assertion always returns false for node 0. The classic encoding returns a counterexample, so it does seem like a stable state exists.

The most likely explanation is that there's a contradiction somewhere in the SMT query which causes it to return UNSAT. Last time I encountered that renaming solved it, but that didn't work when I tried it here. The problem also doesn't occur if the assertion is false for every node. In particular,

`let assert node x = match node with | 0n -> false | _ -> true` erroneously reports no counterexamples, while
`let assert node x = match node with | 0n -> false | _ -> false` correctly reports failures at every node, including node 0.

Testing is hampered by the fact that sometimes (always?) when running unboxed the error in #9 occurs, which makes it hard to tell what's going on.
